GoogleDataTransport/GDTLibrary/GDTStorage.m

```diff
@@ -111,7 +111,14 @@ - (void)storeEvent:(GDTEvent *)event {
 
     // If running in the background, save state to disk and end the associated background task.
     if (bgID != GDTBackgroundIdentifierInvalid) {
+#if TARGET_OS_MACCATALYST
+      NSData *data = [NSKeyedArchiver archivedDataWithRootObject:self
```

Comment:
This is just the new way of doing this and `archiveRootObject:toFile:` is now deprecated. It seems like this should be the approach on iOS 11+, macOS 10.13+, and on Catalyst.

Reply:
Yeah, since [[NSKeyedArchiver archiveRootObject:toFile:]](https://developer.apple.com/documentation/foundation/nskeyedarchiver/1410621-archiverootobject?language=objc) is deprecated, I would suggest using `if (@available(...))` instead of `#if TARGET_OS_MACCATALYST`. An option how it can be done is available [here](https://github.com/firebase/firebase-ios-sdk/blob/92cd10913f6adb82f19b6b4028c9b358ed4a3351/FirebaseInstallations/Source/Library/FIRSecureStorage.m#L193).